### PR TITLE
Fix sampleExample with example method name

### PR DESCRIPTION
**Pull request by Ducasse**
fixed the broken behavior of having selector = example blocking the inspection of example when use <sampleInstance>

### CODE REVIEW

**Comment by dionisiydk** [comment]
Two questions:

1. Why you rename executeBy to executedBy? It does not look right 

2. Are you sure that methods with example pragma should open inspector? It was not in Nautilus. There are many examples which open windows or morphs (FTExamples class>>example1, CheckBoxExample class>>example)

**Comment by dionisiydk** [comment]
I guess what you want is <sampleInstance> pragma be in more priority than method beginning with word "example"? 
I did not saw this case

**Comment by Ducasse** [comment]
Denis
1- ExecuteBy is not english!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
2- <sampleInstance> should always open an inspector even if the method is named example.
3- <example> should open an inspector on the result like that we can play with it.

**Comment by Ducasse** [comment]
so you can rename executedBy: by something else like executeWith: or executeOn: but not executeBy:

**Comment by Ducasse** [comment]
which one do you prefer?

**Comment by dionisiydk** [comment]
I checked Nautilus. It was not really intentionally done. So it is not something which I skip.
In Nautilus it was two actions:  MethodIsExampleAction and MethodSampleInstanceAction. Both with equal actionOrder. So only by incident it was working like you expected.

**Comment by Ducasse** [comment]
Now I do not see why executedBy: does not work.

execute
	script executedBy: targetClass

**Comment by dionisiydk** [comment]
About name I would prefer executeOn: if executeBy is bad.

**Comment by Ducasse** [comment]
Ok I will commit it.

**Comment by Ducasse** [comment]
About nautilus I do not really care how it was implemented. I care that as a user it works.

**Comment by Ducasse** [comment]
Ok now you should get executeOn:

**Comment by dionisiydk** [comment]
But basic logic of methods without <sampleInstance> should not open inspector. Can you modify example script method with extra condition?
```Smalltalk
ClyExampleScript class>>#isImplementedByMethod: aMethod
     (aMethod hasPragmaNamed: #sampleInstance) ifTrue: [^false].
	^(aMethod selector beginsWith: 'example') 
		or: [aMethod hasPragmaNamed: #example]
```

**Comment by Ducasse** [comment]
But basic logic of methods without should not open inspector. 

I do not understand the sentence.
Do you mean 
But basic logic of methods without <sampleInstance> should not open inspector. 

Can you modify example script method with extra condition?

Ok I will

**Comment by dionisiydk** [comment]
I edit my message. It was markdown which hide sampleInstance word

**Comment by Ducasse** [comment]
So in that case I should revert the inspect on the example. But I think that opening an inspector is nice. 
I do not get why we have example and script if example does not open an inspector.

**Comment by Ducasse** [comment]
I will remove the inspect from example. I still do not get what is the difference between script and example.

**Comment by dionisiydk** [comment]
I am ok with all of it. Just point you how it was before and that there are many examples in image where inspector does not make any sense

File: src/Calypso-SystemPlugins-ClassScripts-Queries/ClyEvaluationScript.class.st [hunk 1]
```diff
@@ -9,5 +9,5 @@ Class {
 
 { #category : #'script detection' }
 ClyEvaluationScript class >> isImplementedByMethod: aMethod [
-	^aMethod hasPragmaNamed: #script
+	^ aMethod hasPragmaNamed: #script
```
Comment (dionisiydk) [inline src/Calypso-SystemPlugins-ClassScripts-Queries/ClyEvaluationScript.class.st]: Do you know why this file is marked with red here?

Comment (Ducasse) [reply src/Calypso-SystemPlugins-ClassScripts-Queries/ClyEvaluationScript.class.st]: No idea.

**Comment by Ducasse** [comment]
Yes I see. I reverting the executeOn:. Now I remove the inspect on example. And if people do not like it then they will have to change <sampleInstance> to <script> or <example>

**Comment by Ducasse** [comment]
I improved the class comments too.

**Comment by Ducasse** [comment]
Ok now my latest changes are pushed.

**Pull request merged**